src/xclim/indices/_threshold.py

```diff
@@ -3633,3 +3638,148 @@ def wet_spell_max_length(
         resample_before_rl=resample_before_rl,
         **indexer,
     )
+
+
+@declare_units(
+    snd="[length]",
+    prsn="[precipitation]",
+    snd_thresh="[length]",
+    prsn_thresh="[length]",
+)
+def holiday_snow_days(
+    snd: xarray.DataArray,
+    snd_thresh: Quantified = "20 mm",
+    op: str = ">=",
+    date_start: str = "12-25",
+    date_end: str | None = None,
+    freq: str = "YS",
+) -> xarray.DataArray:  # numpydoc ignore=SS05
+    """
+    Christmas Days.
+
+    Whether there is a significant amount of snow on the ground on December 25th (or a given date range).
+
+    Parameters
+    ----------
+    snd : xarray.DataArray
+        Surface snow depth.
+    snd_thresh : Quantified
+        Threshold snow amount. Default: 20 mm.
+    op : {">", "gt", ">=", "ge"}
+        Comparison operation. Default: ">=".
+    date_start : str
+        Beginning of analysis period. Default: "12-25" (December 25th).
+    date_end : str, optional
+        End of analysis period. If not provided, `date_start` is used.
+        Default: None.
+    freq : str
+        Resampling frequency. Default: "YS".
+        The default value is chosen for the northern hemisphere.
+
+    Returns
+    -------
+    xarray.DataArray, [bool]
+        Boolean array of years with Christmas Days.
+
+    References
+    ----------
+    https://www.canada.ca/en/environment-climate-change/services/weather-general-tools-resources/historical-christmas-snowfall-data.html
+    """
+    snd_constrained = select_time(
+        snd,
+        drop=True,
+        date_bounds=(date_start, date_start if date_end is None else date_end),
+    )
+
+    xmas_days = count_occurrences(
+        snd_constrained, snd_thresh, freq, op, constrain=[">=", ">"]
+    )
+
+    xmas_days = xmas_days.assign_attrs({"units": "days"})
```

Comment:
```suggestion
    xmas_days = to_agg_units(xmas_days, snd,  "count")
```
This will put the correct units, depending on the sampling frequency of the data. I know your function assumes daily, but the explicit check is only done on the indicator. Thus, the function should still be able to manage other samplings.